neo4j/_async/io/_bolt.py

```diff
@@ -746,18 +755,19 @@ async def close(self):
         finally:
             self._closed = True
 
-    async def close_non_blocking(self):
```

Comment:
this is a public method, is there any other client for this method?

Reply:
While `close_non_blocking` is public, the connection class is never exposed to the user. So there is no risk of breaking user-code. All driver-internal usages of `close_non_blocking` have been replace with `kill` which is the proper way of handling cancellations. I was also used to clean up failed or timed out connection attempts (error/timeout while sending/waiting for response to the `HELLO` message). I think it's also fine to forcefully close the connection in such cases.